### PR TITLE
Change interestPeriod for a reservation

### PR DESCRIPTION
**Pull request by kasperbirch1**
#### This PR adds the possibility to change the InterestPeriod for a reservation 

The following is added: 
- NoInterestAfterModal
- getFutureDateString 
Returns a string (YYYY-MM-DD)
- move getNoInterestAfter to reservation/helper.ts
- add translations 

#### Screenshot of the result
Skærmbillede 2022-09-22 kl  14 29 39
Skærmbillede 2022-09-22 kl  14 29 45
Skærmbillede 2022-09-22 kl  14 30 37

#### Checklist

- [ ] My complies with [our coding guidelines](../docs/code_guidelines.md).
- [ ] My code passes our static analysis suite. If not then I have added a comment explaining why this change should be exempt from the code standards and process.
- [ ] My code passes our continuous integration process. If not then I have added a comment explaining why this change should be exempt from the code standards and process.

#### Additional comments or questions

If you have any further comments or questions for the reviewer them please add them here.

### CODE REVIEW

File: src/components/reservation/helper.ts [hunk 1]
```diff
@@ -9,4 +10,37 @@ export const getPreferredBranch = (id: string, array: AgencyBranch[]) => {
...
+    "30": t("oneMonthText"),
+    "60": t("twoMonthsText"),
+    "90": t("threeMonthsText"),
+    "180": t("sixMonthsText"),
+    "360": t("oneYearText")
+  };
+};
+
+export const getNoInterestAfter = (
+  days: number | string,
```
Comment (kasperg) [inline src/components/reservation/helper.ts:24]: Please look into how we can limit the types we access for our internal functions. A number of days should only be in the form of a number - not a string which may contain a number or not.

The fact that JavaScript object cannot have numeric keys is an implementation detail.

File: src/components/reservation/helper.ts [hunk 2]
```diff
@@ -9,4 +10,37 @@ export const getPreferredBranch = (id: string, array: AgencyBranch[]) => {
...
+    reservationInterestIntervals.default
+  );
+};
+
+export const getFutureDateString = (num: string | number) => {
+  const today = new Date();
+  const futureDate = new Date();
+  futureDate.setDate(today.getDate() + Number(num));
+  return futureDate.toISOString().slice(0, 10);
+};
```
Comment (kasperg) [inline src/components/reservation/helper.ts:44]: Please rework this function.

Our internal representation of the users interest date should be a number of days wherever possible. Not a string.

When working with dates where the standard JavaScript date api does not help us much (outputting a date in a specific format, modifying a date) we use `Day.js`. It provides helpful methods e.g. [`format()`] and [`add()'](https://day.js.org/docs/en/manipulate/add).

File: src/components/reservation/reservation-modal.tsx [hunk 3]
```diff
@@ -54,6 +54,7 @@ const ReservationModal = ({
   const [reservationResponse, setReservationResponse] =
     useState<ReservationResponseV2 | null>(null);
   const [selectedBranch, setSelectedBranch] = useState<string | null>(null);
+  const [selectedInterest, setSelectedInterest] = useState<string | null>(null);
```
Comment (kasperg) [inline src/components/reservation/reservation-modal.tsx:57]: This is probably where we want to ensure that the interest day is a number not a string.

File: src/components/reservation/forms/ModalReservationFormSelect.tsx [hunk 4]
```diff
@@ -12,7 +11,7 @@ export interface ModalReservationFormSelectProps {
     title: string;
     description: string[];
   };
-  items: AgencyBranch[];
+  items: { label: string; value: string }[];
```
Comment (kasperg) [inline src/components/reservation/forms/ModalReservationFormSelect.tsx:14]: :+1: Nice refactor.

**Review by kasperg — CHANGES_REQUESTED** [review]
:+1: Good stuff here. Please look into your type handling for interest dates.

**Review by kasperg — APPROVED** [review]
:+1: LGTM

**Review by sinejespersen — APPROVED** [review]

**Pull request merged**